src/components/ImageViewerContainer.ts

```diff
@@ -0,0 +1,138 @@
+type Units = "auto" | "pixels" | "percentage";
+
+class ImageViewerContainer extends Component<ImageViewerContainerProps, ImageViewerContainerState> {
+    private subscriptionHandle: number;
+    private attributeCallback: (mxObject: mendix.lib.MxObject) => () => void;
+
+    constructor(props: ImageViewerContainerProps) {
+        super(props);
+
+        const alertMessage = ImageViewerContainer.validateProps(props);
+        this.state = {
+            alertMessage,
+            imageUrl: "",
+            isLoading: true
+        };
+        this.attributeCallback = mxObject => () => this.getImageUrl(mxObject);
+    }
+
+    render() {
+        if (this.state.alertMessage) {
+            return createElement(Alert, { message: this.state.alertMessage });
+        }
+        if (this.state.isLoading) {
+            return DOM.div(null,
+                DOM.i({ className: "glyphicon glyphicon-cog glyph-spin" }),
+                DOM.span(null, " Loading ...")
+            );
+        }
+
+        return createElement(ImageViewer, {
+            height: this.props.height,
+            heightUnits: this.props.heightUnit,
+            imageurl: this.state.imageUrl,
+            width: this.props.width,
+            widthUnits: this.props.widthUnit
+
+        });
+    }
+
+    componentWillReceiveProps(newProps: ImageViewerContainerProps) {
+        this.resetSubscriptions(newProps.mxObject);
+        this.setState({ isLoading: true });
+        this.getImageUrl(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        if (this.subscriptionHandle) window.mx.data.unsubscribe(this.subscriptionHandle);
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        if (this.subscriptionHandle) window.mx.data.unsubscribe(this.subscriptionHandle);
+
+        if (mxObject) {
+            this.subscriptionHandle = window.mx.data.subscribe({
+                callback: this.attributeCallback(mxObject),
+                guid: mxObject.getGuid()
+            });
+        }
+    }
+
+    public static validateProps(props: ImageViewerContainerProps): string {
+        let message = "";
+        if (props.source === "urlAttribute" && !props.imageAttribute) {
+            message = "Configuration error: for data source Dynamic URL; Dynaic URL attribute is required";
+        }
+        if (props.source === "staticUrl" && !props.urlStatic) {
+            message = "Configuration error: for data source Static URL; a static url is required";
+        }
+        if (props.source === "staticImage" && !props.imageStatic) {
+            message = "Configuration error: for data source Static Image; a static image is required";
+        }
+
+        return message;
+    }
+
+    private getImageUrl(mxObject?: mendix.lib.MxObject) {
+        if (mxObject && this.props.source === "urlAttribute") {
+            this.setState({
+                imageUrl: mxObject.get(this.props.imageAttribute) as string,
```

Comment:
Conditional is checking against urlAttribute but you're retrieving from imageAttribute.